### PR TITLE
ci: support python 3.11

### PR DESCRIPTION
**Pull request by mteiste**
the "standard" pylint action did not work for python 3.11 so replaced it with re-introduced `make pylint`.
Until full cleanup has been made only fail the build if pylint step finds Errors.
Mark the "known" import errors instead of blindly allowing all import errors unlike previously done in the commit hook implementation.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -43,23 +47,24 @@ jobs:
     strategy:
       max-parallel: 4
       matrix:
-        python-version: [3.8, "3.10"]
+        python-version: ["3.9", "3.10", "3.11"]
```
Comment (fingon) [inline .github/workflows/build.yml:50]: Could probably drop 3.10 here.

File: test/test_websocket.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 import asyncio
 import json
 import logging
-import snappy
+import snappy  # pylint: disable=import-error
```
Comment (fingon) [inline test/test_websocket.py:9]: Why is this needed?

File: Makefile [hunk 3]
```diff
@@ -10,6 +11,10 @@ PYLINT_DIRS = journalpump/ test/ systest/
 unittest:
 	$(PYTHON) -m pytest -vv test/
 
+.PHONY: pylint
+pylint:
+	$(PYTHON) -m pylint --rcfile .pylintrc $(PYLINT_DIRS) || exit $$(($$?&2))  # only fail on errors
```
Comment (fingon) [inline Makefile:16]: `pylint --enable F,E` ?

Fatal, error; warning is enabled by default too I think.

I would probably define this CI-only target to be error only - running the W and I checks and ignoring their results in CI doesn't make much sense (and default pylinting presumably is done elsewhere).

**Review by fingon — CHANGES_REQUESTED** [review]

File: Makefile (hunk 3 above)
Comment (mteiste) [reply Makefile:16]: because (tried) the --enable F,E has no effect on the return status

File: test/test_websocket.py (hunk 2 above)
Comment (mteiste) [reply test/test_websocket.py:9]: most likely for the same reason the pre-commit hook mypy has `args: [--ignore-missing-imports]` and the origin pre-commit hook pylint had `[--disable=E0401]`, to be honest i did not investigate further why the snappy is not available when the pre-commit hooks are run.

**Review by fingon — APPROVED** [review]
